### PR TITLE
Add tests for collection views

### PR DESCRIPTION
**Pull request by lojals**
#### What does this PR do? Any background context you want to provide?
This PR adds the missing unit tests for empty states on collection views.
#### Where should the reviewer start?
* `Example/Tests/DataSourceTests.swift`
#### How should this be manually tested?
* Run the tests.
#### What are the relevant tickets?
https://github.com/wizeline/WLEmptyState/issues/26

### CODE REVIEW

File: WLEmptyState/Classes/Views/EmptyStateView.swift [hunk 1]
```diff
@@ -86,6 +86,7 @@ final class EmptyStateView: UIView {
     }
     
     override public func didMoveToSuperview() {
+        super.didMoveToSuperview()
```
Comment (leonardo-garcia) [inline WLEmptyState/Classes/Views/EmptyStateView.swift]: Nice catch 👌

File: Example/Tests/UICollectionViewMocks.swift [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+//  UICollectionViewMocks.swift
+//  WLEmptyState_Example
+//
+//  Created by Jorge Ovalle on 6/27/19.
+//
+
+import Foundation
+import WLEmptyState
+
+class DefaultCollectionViewController: UICollectionViewController, WLEmptyStateDataSource {
```
Comment (leonardo-garcia) [inline Example/Tests/UICollectionViewMocks.swift]: Not a big deal. Though, can we mark these classes as `final?

File: WLEmptyState/Classes/Protocols/WLEmptyStateProtocol.swift [hunk 3]
```diff
@@ -47,30 +47,23 @@ extension WLEmptyStateProtocol {
...
-    
-    var numberOfItems: Int {
-        var items = 0
-        let numberOfSections = dataSource?.numberOfSections?(in: self) ?? 1
-        for section in 0..<numberOfSections {
-            items += dataSource?.collectionView(self, numberOfItemsInSection: section) ?? 0
-        }
-        return items
+    private func getNumberOfSections() -> Int {
+        return (self as? UITableView).flatMap { $0.dataSource?.numberOfSections?(in: $0) } ??
```
Comment (Abel7x) [inline WLEmptyState/Classes/Protocols/WLEmptyStateProtocol.swift]: 👌
